### DISCUSSION TITLE
Naming

### DISCUSSION THREAD

**Discussion by MadAlex1997** [Mojo-Numerics-and-Algorithms-group/NuMojo#8] [General]
Before we make an organization I would appreciate feedback on possible names for this project. I rather like NuMojo, but it is only fair to get a community consensus. I will give it one week (deadline11:59 pm19 May 2024) then I will put our top names in a poll on the Mojo discord.

# Rules
* Name suggestions must reference the primary function of the library (numerics, scientific computing, etc)
* Name suggestions must not overlap with existing Python libraries, and preferably not overlap with any major software product or project (for the sake of searchability)
* Name suggestions must not resemble any derogatory word (if you speak a language in which a suggestion is vulgar please reply to that suggestion with a description of why it is vulgar and in what language)
* Name suggestions should include an abbreviated form with no more than 10 alphanumeric characters (this is the name that will be used for importing the library, think sklearn for scikit-learn) this abbreviation should also make it obvious what the library is for as this will likely be used in place of the library name in common conversation and documentation
* I reserve the right to veto names and workshop names and abbreviations collectively with the community, I don't intend to be a dictator on this, but I want the name to be professional.
* The top five voted names plus the original name NuMojo will be put up on a poll on Discord on Monday 20th of May. The winner of that poll will be the name of the project and Git Hub organization.

**Comment by MadAlex1997** [comment]
To get this started

# Name
NuMojo
# Abbreivation
NuMojo or numojo depending on if we prefer all lower case import names
# Description
Shortend form of Numerical Mojo, straightforward and easily recognizable, has already been mentioned in Modular's weekly newsletter.

**Reply by ndrewGele** [reply]
I'm a fan. 
To play devil's advocate, it could be interpreted as "New Mojo", but honestly I can't think of any alternatives.

**Reply by MadAlex1997** [reply]
@ndrewGele 

Yeah, I have considered adding a second m so that it would be NumMojo, but in library lower format nummojo looks goofy, like baby talk.

In my head, I pronounce it like Num-Ojo with an elongated m.

**Reply by AzizAlqasem** [reply]
Good, but can the suffix jo be omitted to be either NuMo or MoNu? The Mo should be good enough to relate to Mojo. NuMojo sounds to me like Numpython instead of Numpy.

**Reply by MadAlex1997** [reply]
@AzizAlqasem I think in python we have a tendency to choose overly short names for actual packages, especially for how easy it is to alias libraries in python. A counter example is seaborn and matplotlib which both have longer names  but have very common abbreviations 
```python
import matplotlib.pyplot as plt
import seaborn as sns
```
Are the most common aliases for those libraries, mostly as a result of being used in the documentation examples.
I could imagine
```mojo
import numojo as nm
```
Being the most common to grab everything and all the top level functions

I could also easily imagine calling specific sub-libraries, as is seen in scipy, also being common for a library of this scope
```mojo
import numojo.optimizers as opt
import numojo.partialdiffeq as pdeq
```

As for why I wish to include the whole name of the language:
* It is four letters for a six letter total full package name
* Mojo files are `.mojo` while Python files are `.py` so if NumPy is `num.py` then NuMojo is `num.mojo` (see the above reply for why I don't like nummojo)
* It is more search-friendly for a young language

**Reply by Moosems** [reply]
I'd also note that python has set the convention of .py in many places including python files but I've never really encountered `mo` in reference to Mojo

**Reply by AzizAlqasem** [reply]
Okey.
Maybe my tongue will get used to it eventually :)
Or I could pronounce it as Nojo instead :)

**Comment by AzizAlqasem** [comment]
# Name
MoNA

# Abbreivation
mona and used as:
```Python
import mona as ma 
ma.sin(x)
```
`ma` is memorable as short of mojo analysis, mojo algorithms, or math.
# Description
* Abbreviated from the Mojo Numerical Analysis/Algorithms library. 
* Its primary function is to cover domains similar to NumPy and SciPy and within the scope of some textbooks in the field such as:
"Numerical Recipes by William H. Press", "Numerical Analysis by Richard L. Burden" and "Numerical Algorithms by Justin Solomon".
* MoNA is easy to remember since it describes the language (MO) and the field (NA).
* MoNA is easy to pronounce!

## Note
NuMojo is a good one, but it emphasizes the language (Mojo) and is less straightforward to pronounce.

**Reply by MadAlex1997** [reply]
My only concern is searchability as it is a common name for a Woman, and interestingly but unimportantly it loosely means immortal or forever (in the sense of God) in Persian, and "wish" in Arabic.

I agree that it is more straightforward to pronounce.

The library is currently findable under the search `numojo` top 10 results on Google and is the third result on Bing. I think a unique name is a benefit. 

Honestly, I hope to see someone post a unique but easier-to-pronounce name that carries the essence of the intended library a little better.

Let's give it a week to get a few more suggestions then put it to a poll.

**Reply by AzizAlqasem** [reply]
@MadAlex1997 It is very likely to find many meanings for an easy-to-pronounce four-letter word :).

Personally, I can live with any name, but I thought it would be great to make sure that the name is as good as what this library is going to be! And deciding the name now is easier than later.
